CacheSim/CacheSimLinux.cpp

```diff
@@ -0,0 +1,575 @@
+/*
+Copyright (c) 2017, Insomniac Games
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+Redistributions of source code must retain the above copyright notice, this
+list of conditions and the following disclaimer.
+
+Redistributions in binary form must reproduce the above copyright notice, this
+list of conditions and the following disclaimer in the documentation and/or
+other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include "Precompiled.h"
+
+#include <asm/prctl.h>
+#include <execinfo.h>
+#include <link.h>
+#include <signal.h>
+#include <sys/auxv.h>
+#include <sys/prctl.h>
+#include <sys/ptrace.h>
+#include <sys/syscall.h>
+#include <sys/user.h>
+#include <sys/wait.h>
+
+struct CONTEXT
+{
+  int64_t Rax;
+  int64_t Rcx;
+  int64_t Rdx;
+  int64_t Rbx;
+  int64_t Rsp;
+  int64_t Rbp;
+  int64_t Rsi;
+  int64_t Rdi;
+  int64_t R8;
+  int64_t R9;
+  int64_t R10;
+  int64_t R11;
+  int64_t R12;
+  int64_t R13;
+  int64_t R14;
+  int64_t R15;
+  int64_t Rip;
+};
+
+
+static struct sigaction g_OldSigAction;
+static bool g_SignalHandlerInstalled = false;
+extern "C" void CacheSimRemoveHandler();
+
+static void DebugBreak()
+{
+  if ( g_SignalHandlerInstalled )
```

Comment:
I wish visual studio had some way of specifying style inside a project. So far it looks like there are two issues:
1) Indentation should be two spaces
2) No spaces inside a control flow statement, i.e. "if (foo)" not "if ( foo )"
Are there any others you noticed?